src/mono/wasm/debugger/DebuggerTestSuite/EvaluateOnCallFrameTests.cs

```diff
@@ -1178,47 +1178,67 @@ async Task EvaluateOnCallFrameFail(string call_frame_id, params (string expressi
                 var testRootHiddenProps = await GetProperties(testRootHidden["objectId"]?.Value<string>());
 
-                var (refList, _) = await EvaluateOnCallFrame(id, "testPropertiesNone.list");
-                var refListProp = await GetProperties(refList["objectId"]?.Value<string>());
-                var list = refListProp.First(v => v["name"]?.Value<string>() is "Items" or "_items");
-                var refListElementsProp = await GetProperties(list["value"]["objectId"]?.Value<string>());
-
-                var (refArray, _) = await EvaluateOnCallFrame(id, "testPropertiesNone.array");
-                var refArrayProp = await GetProperties(refArray["objectId"]?.Value<string>());
-
-                var (refStruct, _) = await EvaluateOnCallFrame(id, "testPropertiesNone.sampleStruct");
-                var refStructProp = await GetProperties(refStruct["objectId"]?.Value<string>());
-
-                var (refClass, _) = await EvaluateOnCallFrame(id, "testPropertiesNone.sampleClass");
-                var refClassProp = await GetProperties(refClass["objectId"]?.Value<string>());
-
-                int refItemsCnt = refListElementsProp.Count() + refArrayProp.Count() + refStructProp.Count() + refClassProp.Count();
-                Assert.Equal(refItemsCnt, testRootHiddenProps.Count());
+                JObject[] expectedListRootHiddenElements = new[]
+                {
+                    JObject.FromObject(new { value = TNumber(1), name = "listRootHidden[0]"}),
+                    JObject.FromObject(new { value = TNumber(2), name = "listRootHidden[1]"})
+                };
+                JObject[] expectedArrayElements = new[]
+                {
+                    JObject.FromObject(new { value = TNumber(11), name = "arrayRootHidden[0]"}),
+                    JObject.FromObject(new { value = TNumber(22), name = "arrayRootHidden[1]"})
+                };
+                JObject[] expectedStructRootHiddenElements = new[]
+                {
+                    JObject.FromObject(new { value = TNumber(100), name = "sampleStructRootHidden.Id"}),
+                    JObject.FromObject(new { value = TBool(true), name = "sampleStructRootHidden.IsStruct"})
+                };
+                JObject[] expectedClassRootHiddenElements = new[]
+                {
+                    JObject.FromObject(new { value = TNumber(200), name = "sampleClassRootHidden.ClassId"}),
+                    JObject.FromObject(new { value = TObject("System.Collections.Generic.List<string>", description: "Count = 1"), name = "sampleClassRootHidden.Items"})
+                };
 
-                //in Console App names are in []
-                //adding variable name to make elements unique
-                foreach (var item in refListElementsProp)
+                // in Console App names are in []
+                // adding variable name to make elements unique
```

Comment:
Correct me if I'm wrong, but we are doing this only because the existing code in the proxy, and tests kinda assumes that the names will be unique in the get-properties results. Is there an open issue about that?

Reply:
It's true, sending two elements with the same name resulted in an error. There was no issue about it, I made this one - https://github.com/dotnet/runtime/issues/77965.